server_fn/src/lib.rs

```diff
@@ -379,54 +427,94 @@ where
     };
 
     let content_type_header = match &enc {
-        Encoding::Url => "application/x-www-form-urlencoded",
+        Encoding::Url | Encoding::GetJSON | Encoding::GetCBOR => {
+            "application/x-www-form-urlencoded"
+        }
         Encoding::Cbor => "application/cbor",
     };
 
     let accept_header = match &enc {
-        Encoding::Url => "application/x-www-form-urlencoded",
-        Encoding::Cbor => "application/cbor",
+        Encoding::Url | Encoding::GetJSON => {
+            "application/x-www-form-urlencoded"
+        }
+        Encoding::Cbor | Encoding::GetCBOR => "application/cbor",
     };
 
     #[cfg(target_arch = "wasm32")]
-    let resp = match args_encoded {
-        Payload::Binary(b) => {
-            let slice_ref: &[u8] = &b;
-            let js_array = js_sys::Uint8Array::from(slice_ref).buffer();
-            gloo_net::http::Request::post(url)
+    let resp = match &enc {
+        Encoding::Url | Encoding::Cbor => match args_encoded {
+            Payload::Binary(b) => {
+                let slice_ref: &[u8] = &b;
+                let js_array = js_sys::Uint8Array::from(slice_ref).buffer();
+                gloo_net::http::Request::post(url)
+                    .header("Content-Type", content_type_header)
+                    .header("Accept", accept_header)
+                    .body(js_array)
+                    .send()
+                    .await
+                    .map_err(|e| ServerFnError::Request(e.to_string()))?
+            }
+            Payload::Url(s) => gloo_net::http::Request::post(url)
                 .header("Content-Type", content_type_header)
                 .header("Accept", accept_header)
-                .body(js_array)
+                .body(s)
                 .send()
                 .await
-                .map_err(|e| ServerFnError::Request(e.to_string()))?
-        }
-        Payload::Url(s) => gloo_net::http::Request::post(url)
-            .header("Content-Type", content_type_header)
-            .header("Accept", accept_header)
-            .body(s)
-            .send()
-            .await
-            .map_err(|e| ServerFnError::Request(e.to_string()))?,
+                .map_err(|e| ServerFnError::Request(e.to_string()))?,
+        },
+        Encoding::GetCBOR | Encoding::GetJSON => match args_encoded {
+            Payload::Binary(_) => panic!(
```

Comment:
Checking out different combinations I ran into this and was a little surprised. I would have expected that `GetCBOR` would serialize the arguments as a URL-encoded string and the response as CBOR. As it stands, is there any scenario in which `GetCBOR` doesn't panic?

Reply:
As far as I understand it, the Payload defined here is the Request Payload from the client, which cannot be binary because it's going into the query string.